react/features/recording/components/RecordingLabel.native.js

```diff
@@ -6,52 +6,34 @@ import { connect } from 'react-redux';
 import { translate } from '../../base/i18n';
 import { CircularLabel } from '../../base/label';
 import { JitsiRecordingConstants } from '../../base/lib-jitsi-meet';
-import { combineStyles } from '../../base/styles';
 
 import AbstractRecordingLabel, {
-    type Props as AbstractProps,
-    _abstractMapStateToProps
+    type Props,
+    _mapStateToProps
 } from './AbstractRecordingLabel';
 import styles from './styles';
 
-type Props = AbstractProps & {
-
-    /**
-     * Style of the component passed as props.
-     */
-    style: ?Object
-};
-
 /**
  * Implements a React {@link Component} which displays the current state of
  * conference recording.
  *
  * @extends {Component}
  */
-class RecordingLabel extends AbstractRecordingLabel<Props, *> {
+class RecordingLabel extends AbstractRecordingLabel<Props> {
 
     /**
-     * Implements React {@code Component}'s render.
+     * Renders the platform specific label component.
      *
      * @inheritdoc
      */
-    render() {
-        const { _visible, mode, style, t } = this.props;
-
-        if (!_visible) {
-            return null;
-        }
-
-        let labelKey;
+    _renderLabel() {
         let indicatorStyle;
 
-        switch (mode) {
+        switch (this.props.mode) {
         case JitsiRecordingConstants.mode.STREAM:
-            labelKey = 'recording.live';
             indicatorStyle = styles.indicatorLive;
             break;
         case JitsiRecordingConstants.mode.FILE:
```

Comment:
Do you think a future iteration of recording should split the label into file and stream labels?

Reply:
It is split now, the logic is in the abstract class. Or is it what you meant?